jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/InputStreamContentSource.java

```diff
@@ -16,24 +16,42 @@
 import java.io.InputStream;
 import java.nio.ByteBuffer;
 
+import org.eclipse.jetty.io.ByteBufferPool;
 import org.eclipse.jetty.io.Content;
+import org.eclipse.jetty.io.NullByteBufferPool;
 import org.eclipse.jetty.util.IO;
 import org.eclipse.jetty.util.thread.AutoLock;
 import org.eclipse.jetty.util.thread.SerializedInvoker;
 
+/**
+ * A {@link Content.Source} that is backed by an {@link InputStream}.
+ * <p>
+ * Data is read from the {@link InputStream} into a buffer that is optionally acquired
+ * from a {@link ByteBufferPool}, and converted to a {@link Content.Chunk} that is
```

Comment:
You should specify where the buffer pool is coming from when none is given during construction.